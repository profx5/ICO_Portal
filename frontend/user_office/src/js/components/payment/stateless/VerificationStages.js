import React from 'react'
import styled from 'styled-components';
import {media} from 'js/utils/media';


const VerificationStages = ({stages, boundSections, stageClickHandler}) => {
    return (
        <List>
            <ListItem onClick={stageClickHandler} data-bind-to={boundSections[0]} className="active">{stages[0]}</ListItem>
            <ListItem onClick={stageClickHandler} data-bind-to={boundSections[1]}>{stages[1]}</ListItem>
            {stages[2] && <ListItem onClick={stageClickHandler} data-bind-to={boundSections[2]}>{stages[2]}</ListItem>}
        </List>
    )
}


export default VerificationStages;

const List = styled.ul`
    border-radius: 4px;
    background: white;
    width: 315px;
    box-shadow: 0 2px 9px 0 rgba(0, 0, 0, 0.03);
    border-radius: 6px;
    overflow: hidden; 
    margin-bottom: 25px;
    ${media.smMinus} {
        display: none;
    }
    @media (max-width: 1230px) {
        width: 260px;
    }
`;

const ListItem = styled.li`
    height: 56px;
    font-size: 16px;
    color: #323c47;
    border-left: 2px solid transparent;
    display: flex;
    align-items: center;
    padding: 0 25px;
    background: transparent;
    position: relative;
    cursor: pointer;
    &:not(:last-child) {
        border-bottom: 2px solid rgba(151,151,151,.2);
    }
    &:first-child {
        border-top-left-radius: 6px;
        border-top-right-radius: 6px;
    }
    &:last-child {
        border-bottom-left-radius: 6px;
        border-bottom-right-radius: 6px;
    }
    &.active {
      color: #3172fd;
      border-left-color: #3172fd;
    }
`;
