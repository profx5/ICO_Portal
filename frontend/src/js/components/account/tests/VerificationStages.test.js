import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import VerificationStages from 'js/components/account/stateless/VerificationStages';


configure({ adapter: new Adapter() });

it('should render stateless VerificationStages', () => {
    const props = {
        boundSections: [],
        stages: []
    }
    const component = shallow(<VerificationStages {...props}/>);
    expect(component).toMatchSnapshot();
})
