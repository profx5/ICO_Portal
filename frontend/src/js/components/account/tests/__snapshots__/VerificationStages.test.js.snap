// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render stateless VerificationStages 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VerificationStages
    boundSections={Array []}
    stages={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef
          className="active"
        />,
        <ForwardRef />,
        undefined,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": undefined,
          "className": "active",
          "data-bind-to": undefined,
          "onClick": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "rules": Array [
              "
    height: 56px;
    color: #323c47;
    border-left: 2px solid transparent;
    display: flex;
    align-items: center;
    padding: 0 25px;
    background: transparent;
    position: relative;
    cursor: pointer;
    &:not(:last-child) {
        border-bottom: 2px solid rgba(151,151,151,.2);
    }
    &:first-child {
        border-top-left-radius: 6px;
        border-top-right-radius: 6px;
    }
    &:last-child {
        border-bottom-left-radius: 6px;
        border-bottom-right-radius: 6px;
    }
    &.active {
      color: #3172fd;
      border-left-color: #3172fd;
    }
",
            ],
          },
          "displayName": "styled.li",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "li",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": undefined,
          "data-bind-to": undefined,
          "onClick": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "rules": Array [
              "
    height: 56px;
    color: #323c47;
    border-left: 2px solid transparent;
    display: flex;
    align-items: center;
    padding: 0 25px;
    background: transparent;
    position: relative;
    cursor: pointer;
    &:not(:last-child) {
        border-bottom: 2px solid rgba(151,151,151,.2);
    }
    &:first-child {
        border-top-left-radius: 6px;
        border-top-right-radius: 6px;
    }
    &:last-child {
        border-bottom-left-radius: 6px;
        border-bottom-right-radius: 6px;
    }
    &.active {
      color: #3172fd;
      border-left-color: #3172fd;
    }
",
            ],
          },
          "displayName": "styled.li",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "li",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      undefined,
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
    border-radius: 4px;
    background: white;
    width: 315px;
    box-shadow: 0 2px 9px 0 rgba(0, 0, 0, 0.03);
    border-radius: 6px;
    overflow: hidden; 
    margin-bottom: 25px;
",
        ],
      },
      "displayName": "styled.ul",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "ul",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef
            className="active"
          />,
          <ForwardRef />,
          undefined,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": undefined,
            "className": "active",
            "data-bind-to": undefined,
            "onClick": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": true,
              "rules": Array [
                "
    height: 56px;
    color: #323c47;
    border-left: 2px solid transparent;
    display: flex;
    align-items: center;
    padding: 0 25px;
    background: transparent;
    position: relative;
    cursor: pointer;
    &:not(:last-child) {
        border-bottom: 2px solid rgba(151,151,151,.2);
    }
    &:first-child {
        border-top-left-radius: 6px;
        border-top-right-radius: 6px;
    }
    &:last-child {
        border-bottom-left-radius: 6px;
        border-bottom-right-radius: 6px;
    }
    &.active {
      color: #3172fd;
      border-left-color: #3172fd;
    }
",
              ],
            },
            "displayName": "styled.li",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "li",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": undefined,
            "data-bind-to": undefined,
            "onClick": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": true,
              "rules": Array [
                "
    height: 56px;
    color: #323c47;
    border-left: 2px solid transparent;
    display: flex;
    align-items: center;
    padding: 0 25px;
    background: transparent;
    position: relative;
    cursor: pointer;
    &:not(:last-child) {
        border-bottom: 2px solid rgba(151,151,151,.2);
    }
    &:first-child {
        border-top-left-radius: 6px;
        border-top-right-radius: 6px;
    }
    &:last-child {
        border-bottom-left-radius: 6px;
        border-bottom-right-radius: 6px;
    }
    &.active {
      color: #3172fd;
      border-left-color: #3172fd;
    }
",
              ],
            },
            "displayName": "styled.li",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "li",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        undefined,
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
    border-radius: 4px;
    background: white;
    width: 315px;
    box-shadow: 0 2px 9px 0 rgba(0, 0, 0, 0.03);
    border-radius: 6px;
    overflow: hidden; 
    margin-bottom: 25px;
",
          ],
        },
        "displayName": "styled.ul",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "ul",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
